From fae292b4a2f9387587ac6e25c4bdd662be6d3cc1 Mon Sep 17 00:00:00 2001
From: Tang Yuantian <Yuantian.Tang@nxp.com>
Date: Thu, 12 Jan 2017 10:19:12 +0800
Subject: [PATCH 353/508] cpufreq: qoriq: enhance bus frequency calculation

device-type property was missed in soc node in dts on some
arm-based platforms. It caused the bus frequency can not be
obtained correctly.
This patch changed the bus frequency calculation method by
looking up clock table to get bus clock and hence get its
frequency.

Signed-off-by: Tang Yuantian <yuantian.tang@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/cpufreq/qoriq-cpufreq.c |   19 ++++++++-----------
 1 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/drivers/cpufreq/qoriq-cpufreq.c b/drivers/cpufreq/qoriq-cpufreq.c
index 007233a..aed4c18 100644
--- a/drivers/cpufreq/qoriq-cpufreq.c
+++ b/drivers/cpufreq/qoriq-cpufreq.c
@@ -54,19 +54,16 @@ struct soc_data {
 
 static u32 get_bus_freq(void)
 {
-	struct device_node *soc;
-	u32 sysfreq;
-
-	soc = of_find_node_by_type(NULL, "soc");
-	if (!soc)
-		return 0;
-
-	if (of_property_read_u32(soc, "bus-frequency", &sysfreq))
-		sysfreq = 0;
+	struct clk *clk;
 
-	of_node_put(soc);
+	clk = clk_get(NULL, "cg-pll0-div1");
+	if (IS_ERR(clk)) {
+		pr_err("%s: can't get bus frequency %ld\n",
+			__func__, PTR_ERR(clk));
+		return PTR_ERR(clk);
+	}
 
-	return sysfreq;
+	return clk_get_rate(clk);
 }
 
 static struct clk *cpu_to_clk(int cpu)
-- 
1.7.5.4

